sdk/formrecognizer/ai-form-recognizer/src/formRecognizerClient.ts

```diff
@@ -0,0 +1,876 @@
+   * @param {string} source Accessible Uri to an Azure Storage Blob container storing the training documents and label files
+   * @param {BeginTrainingOptions} [options] Options to the BeginTraining operation
+   */
+  public async beginTrainingWithLabel(
+    source: string,
+    options: BeginTrainingOptions<LabeledFormModelResponse> = {}
+  ): Promise<PollerLike<PollOperationState<LabeledFormModelResponse>, LabeledFormModelResponse>> {
+    const trainPollerClient: TrainPollerClient<LabeledFormModelResponse> = {
+      getModel: (modelId: string, options: GetModelOptions) =>
+        this.getLabeledModel(modelId, options),
+      trainCustomModelInternal: (
+        source: string,
+        _useLabelFile?: boolean,
+        options?: TrainModelOptions
+      ) => trainCustomModelInternal(this.client, source, true, options)
+    };
+
+    const poller = new BeginTrainingPoller({
+      client: trainPollerClient,
+      source,
+      intervalInMs: options.intervalInMs,
+      onProgress: options.onProgress,
+      resumeFrom: options.resumeFrom,
+      trainModelOptions: options
+    });
+
+    await poller.poll();
+    return poller;
+  }
+
+  /**
+   * Extracts name-value pairs and tables from a given document using a model from unsupervised training.
+   * This method returns a long running operation poller that allows you to wait
+   * indefinitely until the copy is completed.
+   * You can also cancel a copy before it is completed by calling `cancelOperation` on the poller.
+   * Note that the onProgress callback will not be invoked if the operation completes in the first
+   * request, and attempting to cancel a completed copy will result in an error being thrown.
+   *
+   * Example usage:
+   * ```ts
+   *   const path = "./Invoice_6.pdf";
+   *   const readStream = fs.createReadStream(path);
+   *
+   *   const client = new FormRecognizerClient(endpoint, new FormRecognizerApiKeyCredential(apiKey));
+   *   const poller = await client.beginExtractForms(modelId, readStream, "application/pdf", {
+   *     onProgress: (state) => { console.log(`status: ${state.status}`); }
+   *   });
+   *   await poller.pollUntilDone();
+   *   const response = poller.getResult();
+   *   console.log(response.status);
+   * ```
+   * @summary Extracts form information from a given document using unlabeled model.
+   * @param {string} modelId Id of the model to use
+   * @param {FormRecognizerRequestBody} body Input document
+   * @param {contentType} Content type of the input. Supported types are "application/pdf", "image/jpeg", "image/png", and "image/tiff";
+   * @param {BeginExtractFormsOptions} [options] Options to the BeginExtractForms operation
+   */
+  public async beginExtractForms(
+    modelId: string,
+    body: FormRecognizerRequestBody,
+    contentType?: ContentType,
+    options: BeginExtractFormsOptions = {}
+  ): Promise<FormPollerLike> {
+    if (!modelId) {
+      throw new RangeError("Invalid model id");
```

Comment:
Should this be a `RangeError`?

Reply:
Yes, we use RangeError in JS for invalid arguments